src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,231 +82,231 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
 	                    if (addr.Default_Address__c) {
 	                       addr.Latest_Start_Date__c = System.today();
 	                       addr.Latest_End_Date__c = null;
 	                    } else {
+	                       UTIL_Debug.debug('****Address updated to not default');
 	                       addr.Latest_End_Date__c = System.today();                        
 	                    }
 	                }
+	                i++;
+		        }
+		        alreadyRunBeforeUpdate = true;
+	        }
+	    
+	        // AFTER INSERT
+	        if (!alreadyRunAfterInsert && triggerAction == TDTM_Runnable.Action.AfterInsert) {
+	            for (SObject so : listNew) {
+	                Address__c addr = (Address__c)so;
+		            // a new address that is marked default or seasonal needs to propogate to the parent Account
+		            if (addr.Default_Address__c || ADDR_Seasonal_UTIL.isSeasonalAddr(addr)) {
+		                putAddrInMap(addr, mapParentIdAddr);
+		            }
+	            }
+	            
+	            if(mapParentIdAddr.size() > 0) {
+	                refreshCurrentAccAddress(mapParentIdAddr, dmlWrapper, triggerAction);
+	                processDml(dmlWrapper, mapParentIdAddr);
 	            }
-	
-	            // AFTER INSERT
-	            if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
-	                // a new address that is marked default or seasonal needs to propogate to the household
-	                if (addr.Default_Address__c || isSeasonalAddr(addr)) 
-	                    mapAccIdAddr.put(addr.Parent_Account__c, addr);
+	            
+	            // enforce address records created only as children of parents
+	            verifyAddrAccContactChildOnly((List<Address__c>)listNew);
+	            // enforce Seasonal Addresses don't overlap
+	            ADDR_Seasonal_UTIL.verifyAddrSeasonalNoOverlap((List<Address__c>)listNew);
+	            // delete empty addresses
+	            deleteEmptyAddrs(listNew);
+	            
+	            alreadyRunAfterInsert = true;
+	        }
+	        
+	        // AFTER UPDATE
+	        if (!alreadyRunAfterUpdate && triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+	            Integer i = 0;
+	            for (SObject so : listNew) {
+	                Address__c addr = (Address__c)so;
+	                Address__c addrOld = listOld == null? null : (Address__c)listOld[i];
+		            Boolean changed = ADDR_Addresses_UTIL.isAddressChanged(addr, addrOld, true);
+		            Boolean seasonalChanged =  ADDR_Seasonal_UTIL.isSeasonalChanged(addr, addrOld);
+		            Boolean changedToDefault = addr.Default_Address__c && (addr.Default_Address__c != addrOld.Default_Address__c);
+		            
+		            //An address that is marked default, of that had its seasonal info changed needs to propogate to the parent.
+		            if (changedToDefault ||seasonalChanged) {
```

Comment:
does the seasonal change always need to do this work, even if it wasn't current, nor becomes current?  or is it harmless to do so anyways?